### PR TITLE
[android] update AOT profiles

### PR DESCRIPTION
**Pull request by jonathanpeppers**
In #6820 Rachel and I noticed that there were quite a few API changes
that made the `.aotprofile`'s for MAUI out of date.

Looking at the [JIT times report][0] for the .NET Podcast app, I saw a
couple we could address:

```
Total (ms) |  Self (ms) | Method
      2.61 |       2.61 | string:SplitInternal (string,string[],int,System.StringSplitOptions)
      1.57 |       1.57 | System.Number:NumberToString (System.Text.ValueStringBuilder&,System.Number/NumberBuffer&,char,int,System.Globalization.NumberFormatInfo)
      1.52 |       1.52 | System.Number:TryParseInt32IntegerStyle (System.ReadOnlySpan`1<char>,System.Globalization.NumberStyles,System.Globalization.NumberFormatInfo,int&)
```

I added usage of `string.Split()`, `int.Parse()`, and `int.ToString()`
to `CommonMethods.cs`.

[0]: https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/profiling.md#profiling-the-jit-compiler

## Results ##

Testing on a Pixel 5, an average of 10 runs.

.NET Podcast app:

    Average(ms): 796.8
    Std Err(ms): 3.47626875319565
    Std Dev(ms): 10.992927018972
    ---
    Average(ms): 786.8
    Std Err(ms): 2.0591260281974
    Std Dev(ms): 6.51152823843988

`dotnet new maui`:

    Average(ms): 563.7
    Std Err(ms): 3.26955654485436
    Std Dev(ms): 10.3392456204503
    ---
    Average(ms): 545.3
    Std Err(ms): 2.39930545505708
    Std Dev(ms): 7.5872700404471

### CODE REVIEW

File: .nuspec/maui.aotprofile.txt [hunk 1]
```diff
@@ -2793,9 +2802,6 @@ Methods:
 	System.Reflection.Assembly System.Runtime.Loader.AssemblyLoadContext:MonoResolveUsingResolveSatelliteAssembly (intptr,string)
 	System.Reflection.Assembly System.Runtime.Loader.AssemblyLoadContext:ResolveSatelliteAssembly (System.Reflection.AssemblyName)
 	System.Reflection.Assembly System.RuntimeType:get_Assembly ()
-	System.Reflection.Assembly[] System.AppDomain:GetAssemblies ()
```
Comment (jonathanpeppers) [inline .nuspec/maui.aotprofile.txt:2796]: We used to call this? Nice that was removed!

File: .nuspec/maui.aotprofile.txt [hunk 2]
```diff
@@ -5307,12 +5303,6 @@ Methods:
 	void Microsoft.Maui.PropertyMapper`2<Microsoft.Maui.IView, Microsoft.Maui.IViewHandler>:.ctor (Microsoft.Maui.IPropertyMapper[])
 	void Microsoft.Maui.PropertyMapper`2<Microsoft.Maui.IView, Microsoft.Maui.IViewHandler>:Add (string,System.Action`2<Microsoft.Maui.IViewHandler, Microsoft.Maui.IView>)
 	void Microsoft.Maui.PropertyMapper`2<Microsoft.Maui.IView, Microsoft.Maui.IViewHandler>:set_Item (string,System.Action`2<Microsoft.Maui.IViewHandler, Microsoft.Maui.IView>)
-	void Microsoft.Maui.Resource/Animation:.cctor ()
-	void Microsoft.Maui.Resource/Attribute:.cctor ()
-	void Microsoft.Maui.Resource/Drawable:.cctor ()
-	void Microsoft.Maui.Resource/Id:.cctor ()
-	void Microsoft.Maui.Resource/Layout:.cctor ()
-	void Microsoft.Maui.Resource/Style:.cctor ()
```
Comment (jonathanpeppers) [inline .nuspec/maui.aotprofile.txt:5315]: These must be from #4912

File: .nuspec/maui-blazor.aotprofile.txt [hunk 3]
```diff
@@ -2013,13 +2022,13 @@ Methods:
...
+	object Microsoft.Maui.Handlers.ViewHandler`2<Microsoft.Maui.IContentView, Microsoft.Maui.Platform.ContentViewGroup>:Microsoft.Maui.IElementHandler.get_PlatformView ()
 	object Microsoft.Maui.Hosting.Internal.MauiFactory:CreateInstance (Microsoft.Extensions.DependencyInjection.ServiceDescriptor)
 	object Microsoft.Maui.Hosting.Internal.MauiFactory:GetService (Microsoft.Extensions.DependencyInjection.ServiceDescriptor)
 	object Microsoft.Maui.Hosting.Internal.MauiFactory:GetService (System.Type,Microsoft.Extensions.DependencyInjection.ServiceDescriptor,System.Collections.Generic.IEnumerable`1<Microsoft.Extensions.DependencyInjection.ServiceDescriptor>)
 	object Microsoft.Maui.Hosting.Internal.MauiFactory:GetService (System.Type)
-	object Microsoft.Maui.MauiContext/<>c__DisplayClass10_0`1<Android.App.Activity>:<AddWeakSpecific>b__0 ()
-	object Microsoft.Maui.MauiContext/<>c__DisplayClass9_0`1<Android.App.Application>:<AddSpecific>b__0 ()
+	object Microsoft.Maui.MauiContext/<>c__10`1<Android.App.Activity>:<AddWeakSpecific>b__10_0 (object)
+	object Microsoft.Maui.MauiContext/<>c__9`1<Android.App.Application>:<AddSpecific>b__9_0 (object)
```
Comment (mattleibow) [inline .nuspec/maui-blazor.aotprofile.txt:2031]: Nice! This was #6787

**Review by mattleibow — APPROVED** [review]

File: .nuspec/maui.aotprofile.txt (hunk 1 above)
Comment (eerhardt) [reply .nuspec/maui.aotprofile.txt:2796]: What's interesting is that this call still seems to be on the call path:

https://github.com/dotnet/maui/blob/fe7bc04c54ea94781a44805bad81b0c2548f4c86/src/Controls/src/Core/DependencyService.cs#L109-L119

If you look, we are still calling `void Microsoft.Maui.Controls.DependencyService:Initialize ()` in this AOT profile. And  the Initialize method will call `GetASsemblies()`. So there must have been a change that is setting `s_initialized` to `true` before `DependencyService:Initialize` is getting called.

I think `DependencyService:SetToInitialized()` is getting called before `Initialize()`, and thus the bool is getting set, and we no longer scan the assemblies for DependencyServices.

Maybe https://github.com/dotnet/maui/pull/5174 did this? I see that change added a call to `SetToInitialize()` [here](https://github.com/dotnet/maui/pull/5174/files#diff-dde1fc5ff685106920c096cded904fb3b548e261984e2244e241d8470d276bc3R136).

I wonder if that is a problem... are DependencyServices no longer getting registered by default in this code?

cc @PureWeen @mattleibow

Comment (mattleibow) [reply .nuspec/maui.aotprofile.txt:2796]: We manually register the services now. And the assembly scanning is no longer supported or needs to be explicitly done by the user.

https://github.com/dotnet/maui/blob/main/src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs#L145

**Pull request merged**